pkg/operator/observe_config.go

```diff
@@ -102,6 +102,91 @@ func (c ConfigObserver) sync() error {
 	return nil
 }
 
+// observeEtcdEndpoints reads the etcd endpoints from the endpoints object and then manually pull out the hostnames to
+// get the etcd urls for our config. Setting them observed config causes the normal reconciliation loop to run
+func observeEtcdEndpoints(kubeClient kubernetes.Interface, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	etcdURLs := []string{}
+	etcdEndpoints, err := kubeClient.CoreV1().Endpoints(etcdNamespaceName).Get("etcd", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return observedConfig, nil
+	}
+	if err != nil {
+		return observedConfig, err
+	}
+	if etcdEndpoints == nil {
+		return observedConfig, nil
+	}
+	for _, subset := range etcdEndpoints.Subsets {
+		for _, address := range subset.Addresses {
+			etcdURLs = append(etcdURLs, "https://"+address.Hostname+"."+etcdEndpoints.Annotations["alpha.installer.openshift.io/dns-suffix"]+":2379")
+		}
+	}
+	if len(etcdURLs) > 0 {
+		unstructured.SetNestedStringSlice(observedConfig, etcdURLs, "storageConfig", "urls")
+	} else {
+		glog.Warningf("no etcd endpoints found")
+	}
+	return observedConfig, nil
+}
+
+// observeClusterConfig observes CIDRs from cluster-config-v1 in order to populate list of restrictedCIDRs
+func observeClusterConfig(kubeClient kubernetes.Interface, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	clusterConfig, err := kubeClient.CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		glog.Warningf("cluster-config-v1 not found in the kube-system namespace")
+		return observedConfig, nil
+	}
+	if err != nil {
+		return observedConfig, err
+	}
+
+	installConfigYaml, ok := clusterConfig.Data["install-config"]
+	if !ok {
+		return observedConfig, nil
+	}
+	installConfig := map[string]interface{}{}
+	err = yaml.Unmarshal([]byte(installConfigYaml), &installConfig)
+	if err != nil {
+		glog.Warningf("Unable to parse install-config: %s", err)
+		return observedConfig, nil
+	}
+
+	// extract needed values
+	//  data:
+	//   install-config:
+	//     networking:
+	//       podCIDR: 10.2.0.0/16
+	//       serviceCIDR: 10.3.0.0/16
+	restrictedCIDRs := []string{}
+	networking, ok := installConfig["networking"].(map[string]interface{})
+	if !ok {
+		return observedConfig, nil
+	}
+	if cidr := networking["podCIDR"]; cidr != nil {
+		restrictedCIDRs = append(restrictedCIDRs, fmt.Sprintf("%v", cidr))
+	} else {
+		glog.Warningf("No value found for install-config/networking/podCIDR.")
```

Comment:
please link me to the issue about providing status on failed config observation

Reply:
> please link me to the issue about providing status on failed config observation

I still don't see an issue about this.

Reply:
Issue https://github.com/openshift/cluster-kube-apiserver-operator/issues/80 opened.